ingest/indexer/service/blockprocessor/full_indexer_block_process_strategy.go

```diff
@@ -25,6 +27,15 @@ func (f *fullIndexerBlockProcessStrategy) IsFullBlockProcessor() bool {
 
 // ProcessBlock implements commondomain.BlockProcessStrategy.
 func (f *fullIndexerBlockProcessStrategy) ProcessBlock(ctx sdk.Context) (err error) {
+	// Detect syncing
+	isNodesyncing, err := f.nodeStatusChecker.IsNodeSyncing(ctx)
+	if err != nil {
+		return &commondomain.NodeSyncCheckError{Err: err}
+	}
+	if isNodesyncing {
+		return commondomain.ErrNodeIsSyncing
+	}
+
```

Comment:
NOTE TO REVIEWER: This is the key block of code changes in this PR. Other changes are mainly about promoting the node status checker from sqs package to common package.